File: app/src/main/java/org/mifos/mobile/ui/activities/RegistrationActivity.java
package org.mifos.mobile.ui.activities;

import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.LinearLayout;

import androidx.annotation.RequiresApi;

import org.mifos.mobile.R;
import org.mifos.mobile.ui.activities.base.BaseActivity;
import org.mifos.mobile.ui.fragments.RegistrationFragment;

import java.util.Objects;

import butterknife.BindView;
import butterknife.OnClick;

public class RegistrationActivity extends BaseActivity {

    @BindView(R.id.register_linear)
    LinearLayout llLogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registration);
        replaceFragment(RegistrationFragment.newInstance(), false, R.id.container);


    }
    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
    @OnClick(R.id.register_linear)
    public void hideKeyboardregister(View view) {
        InputMethodManager manager = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
        manager.hideSoftInputFromWindow(Objects.requireNonNull(getCurrentFocus())
                .getWindowToken(), 0);
    }
}

File: app/src/main/java/org/mifos/mobile/ui/activities/RegistrationActivity.kt
package org.mifos.mobile.ui.activities

import android.content.DialogInterface
import android.os.Bundle
import org.mifos.mobile.R
import org.mifos.mobile.databinding.ActivityRegistrationBinding
import org.mifos.mobile.ui.activities.base.BaseActivity
import org.mifos.mobile.ui.fragments.RegistrationFragment
import org.mifos.mobile.utils.MaterialDialog

class RegistrationActivity : BaseActivity() {

    private lateinit var binding: ActivityRegistrationBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRegistrationBinding.inflate(layoutInflater)
        setContentView(binding.root)
        replaceFragment(RegistrationFragment.newInstance(), false, R.id.container)
    }

    override fun onBackPressed() {
        MaterialDialog.Builder().init(this)
            .setTitle(getString(R.string.dialog_cancel_registration_title))
            .setMessage(getString(R.string.dialog_cancel_registration_message))
            .setPositiveButton(getString(R.string.yes)) { _, _ -> super.onBackPressed() }
            .setNegativeButton(R.string.no) { dialog, _ -> dialog.dismiss() }
            .createMaterialDialog()
            .show()
    }

}
